### DISCUSSION TITLE
$wire.set()

### DISCUSSION THREAD

**Discussion by lukas-frey** [livewire/livewire#5954] [1. Help] [answered]
In the beta documentation of Livewire 3, it says that there are two ways to update a property from the client side:
```js
$wire.foo = 'bar';
```

and
```js
$wire.set('foo', 'bar');
```

The difference is that the first variant will work without sending a network request and the second will immediately send a network request.

Would it be possible to add a third argument to `$wire.set()` that lets us choose whether a network request should be sent or not? Like `$wire.set(<property>, <value>, <sendNetwork=true>)`. It can even be default true.

The reason for this request is that sometimes, when you need to update a property dynamically (ie the name of the property is stored inside a variable), you have no way to update it without sending a network request.

**Comment by joshhanley** [comment, marked as answer]
@LukasFreyCZ set already has a third property which is bool for defer. Set it to true to defer the request.

But I think it should potentially be changed following the new standard and make set deferred by default and add a Boolean property for live.

**Reply by lukas-frey** [reply]
Oh, sorry didn't realise that. Thanks for the quick answer! :)

**Reply by lukas-frey** [reply]
Hey so I quickly tested it and it turns out it already follows the new standard. The third parameter is `live = true`. Works like a charm.

**Reply by joshhanley** [reply]
@LukasFreyCZ oh sweet! Thanks for checking. If you think the docs need updating PRs are welcome! 🙂
